plugins/pipelines/startAll.js

```diff
@@ -0,0 +1,79 @@
+'use strict';
+
+const boom = require('boom');
+const joi = require('joi');
+const schema = require('screwdriver-data-schema');
+const idSchema = joi.reach(schema.models.pipeline.base, 'id');
+
+module.exports = () => ({
+    method: 'POST',
+    path: '/pipelines/{id}/startall',
+    config: {
+        description: 'Start all child pipelines given a specific pipeline',
+        notes: 'Start all child pipelines given a specific pipeline',
+        tags: ['api', 'pipelines'],
+        auth: {
+            strategies: ['token'],
+            scope: ['user', '!guest']
+        },
+        plugins: {
+            'hapi-swagger': {
+                security: [{ token: [] }]
+            }
+        },
+        handler: (request, reply) => {
+            const pipelineFactory = request.server.app.pipelineFactory;
+            const eventFactory = request.server.app.eventFactory;
+            const userFactory = request.server.app.userFactory;
+            const id = request.params.id;
+            const username = request.auth.credentials.username;
+            const scmContext = request.auth.credentials.scmContext;
+            const scm = pipelineFactory.scm;
+
+            return Promise.all([
+                pipelineFactory.get(id),
+                userFactory.get({ username, scmContext })
+            ])
+                .then(([pipeline, user]) => {
+                    if (!pipeline) {
+                        throw boom.notFound('Pipeline does not exist');
+                    }
+
+                    return user.getPermissions(pipeline.scmUri)
+                        // check if user has push access
+                        .then((permissions) => {
```

Comment:
Are we going to check if the user has permission on the child pipelines? Or is it implied?

Reply:
Don't think we need to check here, it is shared permissions.  e.g. I have permissions for child A, B, you have permission for child C and added that. All child A, B, C should start.

Reply:
I see, so for startall we only care if the user has permissions against the config pipeline. If we did a Start on a childPipeline, we would still check permissions against that child repo.